spec/models/gateway/amazon_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'spec_helper'
+
+describe Spree::Gateway::Amazon do
+  let(:payment_method) { Spree::Gateway::Amazon.create!(name: 'Amazon', preferred_test_mode: true) }
+  let(:user) { create :user }
+  let(:order) { create(:order_with_line_items, state: 'delivery', user: user) }
+  let(:payment_source) { Spree::AmazonTransaction.create!(order_id: order.id, order_reference: 'REFERENCE')}
+  let(:payment) do
+    order.amazon_transactions.create(order_reference: 'ORDER_REFERENCE')
```

Comment:
This is actually not necessary, you only need one amazon transaction per-payment, this will be linked to the payment's order through the `order_id` column on the `AmazonTransaction`